nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -689,6 +693,71 @@ in
           '';
       };
 
+      proxyCache = mkOption {
+        type = types.submodule {
+          options = {
+            enable = mkEnableOption (lib.mdDoc "Enable proxy cache");
+
+            keysZoneName = mkOption {
+              type = types.str;
+              default = "cache";
+              example = "my_cache";
+              description = lib.mdDoc "Set name to shared memory zone.";
+            };
+
+            keysZoneSize = mkOption {
+              type = types.str;
+              default = "10m";
+              example = "32m";
+              description = lib.mdDoc "Set size to shared memory zone.";
+            };
+
+            levels = mkOption {
+              type = types.str;
+              default = "1:2";
+              example = "1:2:2";
+              description = lib.mdDoc ''
+                The levels parameter defines hierarchy levels of a cache: from 1 to 3,
+                each level accepts values 1 or 2.
```

Comment:
This doesn't explain what the difference between 2 and 3 numbers. Can we simply link the upstream doc?

Reply:
Description taken from here - http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_path
```
Sets the path and other parameters of a cache. Cache data are stored in files. The file name in a cache is a result of applying the MD5 function to the [cache key](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_cache_key). The levels parameter defines hierarchy levels of a cache: from 1 to 3, each level accepts values 1 or 2. For example, in the following configuration
```

Reply:
Update